keyboards/reviung39/keymaps/toshi0383/keymap.c

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2019 gtips
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _BASE,
+    _LOWER,
+    _RAISE,
+    _ADJUST
+};
+
+#define LOWER  MO(_LOWER)
+#define RAISE  MO(_RAISE)
+#define ADJUST MO(_ADJUST)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_reviung39(
+   KC_ESC, KC_Q,  KC_W,  KC_E,  KC_R,  KC_T,             KC_Y,  KC_U,  KC_I, KC_O,   KC_P, KC_BSPACE,
+   KC_TAB, KC_A,  KC_S,  KC_D,  KC_F,  KC_G,             KC_H,  KC_J,  KC_K, KC_L,   KC_RGUI, KC_RSFT,
+   KC_SPC, KC_Z,  KC_X,  KC_C,  KC_V,  KC_B,             KC_N,  KC_M,  KC_RCTL, KC_RGUI, KC_RALT, RGUI(KC_RSFT),
+                                       LOWER,  KC_ENT,  RAISE
+  ),
+  
+  [_LOWER] = LAYOUT_reviung39(
+   KC_TILD, KC_EXLM, KC_AT, KC_HASH, KC_DLR,  KC_PERC,    KC_CIRC, KC_AMPR, KC_ASTR, _______, _______, _______,
+   _______, KC_LT, KC_GT, KC_LPRN, KC_RPRN, KC_GRV,       KC_MINS, KC_LCBR, KC_RCBR, KC_LBRC, KC_RBRC, KC_PIPE,
+   _______, _______, _______, _______, KC_EQL, KC_QUES,   KC_UNDS, KC_PLUS, KC_COMM, KC_DOT, KC_SLSH, _______,
+                                      _______,  KC_ENT,   KC_RGUI
+  ),
+  
+  [_RAISE] = LAYOUT_reviung39(
+  _______, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  _______,
+  _______, KC_COLN, KC_SCLN, KC_DQT, KC_QUOT, KC_BSLS,    KC_LEFT, KC_DOWN, KC_UP, KC_RGHT, _______, _______,
+  _______, _______, _______, _______, _______, _______,   ADJUST, _______, _______, _______, _______, RGB_MOD,
```

Comment:
It’s here.

Reply:
You need to delete the `layer_state_set_user` function on Lines 59-61. That function makes it so you can only access the Adjust layer if *both* the Lower and Raise layers are active. That function runs every time you change layers (either on or off). It's checking the layer state, finding that the Lower layer isn't active, and so it keeps you on the Raise layer.

If you delete the function, I think it should work.

Reply:
Thank you so much!
Resolved in https://github.com/qmk/qmk_firmware/pull/7141/commits/53e49b9553ec08fc231cc137ff5f652a1ebaec67